### ISSUE TITLE
[semantic-tokens] Refresh tokens after apply workspaceEdit

### ISSUE THREAD

**Issue by ericdallo** [emacs-lsp/lsp-mode#3155]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command
```bash
emacs -q -l lsp-start-plain.el
```
where `lsp-start-plain.el` can be downloaded [here](https://github.com/emacs-lsp/lsp-mode/blob/master/scripts/lsp-start-plain.el).

### Bug description

We should refresh the semantic tokens of all affecter buffers by a workspaceEdit change sent from server.

### Steps to reproduce

call LSP rename on some symbol that is used on multiple already opened buffers, after renaming the symbol on other buffers will have outdated semantic tokens.

### Expected behavior

refresh the tokens after applying the workspaceEdit

### Which Language Server did you use?

clojure-lsp and dart

### OS

Linux

### Anything else?

c/c @sebastiansturm

**Comment by ericdallo** [comment]
It seems we are sending the request semantic token before we send didChange, causing these outdated tokens c/c @sebastiansturm

**Comment by sebastiansturm** [comment]
I pushed a change to my fork that refreshes tokens on `lsp-after-change`, though I'm still not sure I'm too fond of this change, will try to make up my mind tomorrow. Haven't tested this extensively but at least the clojure-lsp example seems to refontify correctly

**Comment by ericdallo** [comment]
yes, I used this during the week and it works perfectly unless I jack in cider which I still have no idea how to fix that :/
